### PR TITLE
[Product CRUD] Abstract todos

### PR DESCRIPTION
**Pull request by mikejolley**
Handles misc methods in abstract class which were marked todo.

- Downloads
- Ratings

And fixes some things I noticed during coding.

### CODE REVIEW

File: includes/abstracts/abstract-wc-legacy-product.php [hunk 1]
```diff
@@ -661,4 +661,61 @@ public function get_matching_variation( $match_attributes = array() ) {
 		_deprecated_function( 'WC_Product::get_matching_variation', '2.7', 'wc_find_matching_product_variation' );
 		return wc_find_matching_product_variation( $this, $match_attributes );
 	}
+
+	/**
+	 * Returns whether or not we are showing dimensions on the product page.
+	 *  @deprecated 2.7.0 Unused.
```
Comment (justinshreve) [inline includes/abstracts/abstract-wc-legacy-product.php]: Hopefully I'm not just seeing things this time (:P), but it looks like there is an extra space in front.

File: includes/abstracts/abstract-wc-product.php [hunk 2]
```diff
@@ -474,6 +477,24 @@ public function get_height( $context = 'view' ) {
...
 	/**
+	 * Returns formatted dimensions.
+	 *
+	 * @param  $formatted True by default for legacy support - will be false/not set in future versions to return the array only. Use wc_format_dimensions for formatted versions instead.
+	 * @return string|array
+	 */
+	public function get_dimensions( $formatted = true ) {
+		if ( $formatted ) {
+			wc_soft_deprecated_argument( 'WC_Product::get_dimensions', '2.7', '2.8', 'By default, get_dimensions has an argument set to true so that HTML is returned. This is to support the legacy version of the method. To get HTML dimensions, instead use wc_format_dimensions() function. Pass false to this method to return an array of dimensions. This will be the new default behavior in future versions.' );
+			return apply_filters( 'woocommerce_product_dimensions', wc_format_dimensions( $variation->get_dimensions( false ) ), $this );
```
Comment (justinshreve) [inline includes/abstracts/abstract-wc-product.php]: `$variation` -> `$this`

File: includes/class-wc-comments.php [hunk 3]
```diff
@@ -328,6 +328,89 @@ public static function add_comment_purchase_verification( $comment_id ) {
...
+
+		return $average;
+	}
+
+	/**
+	 * Get product review count for a product (not replies). Please note this is not cached.
+	 *
+	 * @since 2.7.0
+	 * @param WC_Product $product
+	 * @return float
```
Comment (justinshreve) [inline includes/class-wc-comments.php]: int

File: includes/wc-deprecated-functions.php [hunk 4]
```diff
@@ -11,7 +11,55 @@
...
+		_deprecated_function( $function, $version, $replacement );
+	}
+}
+
+/**
+ * Soft deprecate an argument so that it's technically deprecated, but not shown
+ * until a future version to ease transition for developers.
+ *
+ * @since  2.7.0
+ * @param  string $function
```
Comment (justinshreve) [inline includes/wc-deprecated-functions.php]: Should be `$argument` instead of `$function`

**Review by emdashcodes — APPROVED** [review]
Left 4 minor comments. Otherwise, I think this can be merged after those are fixed.

**Comment by justinshreve** [comment]
Based on the size of the incoming data store PR, I think it might make sense to hold off and have us rebase this one after it's merged. This has a few changes that will be incompatible, but I think it will be easier to rebase this one then the other.

**Pull request merged**